NUService.js

```diff
@@ -3,21 +3,31 @@ import NURESTConnection from './NURESTConnection';
 import NURESTUser from './NURESTUser';
 import ServiceClassRegistry from './ServiceClassRegistry';
 
+const getURLParams = (url) => {
+    const anchor = document.createElement('a');
```

Comment:
why do you need this? please use Regex on the rootURL!

Reply:
looked around, this seems to be the simplest way to extract URL components in javascript.
also, couldnt come up with an appropriate regex, any regex suggestion?

Reply:
here some code:
```
const url = new URL(rootURL)
const [RESTRoot, RESTResource, APIVersion] = url.pathname.split('/')
const host = url.hostname;
const port = url.port;
```